packages/fxa-content-server/app/scripts/templates/sign_in_password.mustache

```diff
@@ -27,41 +27,50 @@
-      <input class="hidden" required />
+          <div id="tos-pp" class="text-grey-500 my-5 text-xs">
+            {{#isPocketClient}}
+              {{#unsafeTranslate}}
+                By proceeding, you agree to:<br />
+                Pocket’s <a class="link-grey" id="pocket-tos" href="https://getpocket.com/en/tos/">Terms of Service</a> and <a id="pocket-pp" class="link-grey" href="https://getpocket.com/en/privacy/">Privacy Notice</a><br />
+                Firefox’s <a class="link-grey" id="fxa-tos" href="/legal/terms">Terms of Service</a> and <a class="link-grey" id="fxa-pp" href="/legal/privacy">Privacy Notice</a>.
+              {{/unsafeTranslate}}
+            {{/isPocketClient}}
+            {{^isPocketClient}}
+              {{#unsafeTranslate}}By proceeding, you agree to the <a class="link-grey" id="fxa-tos" href="/legal/terms">Terms of Service</a> and <a class="link-grey" id="fxa-pp" href="/legal/privacy">Privacy Notice</a>.{{/unsafeTranslate}}
+            {{/isPocketClient}}
+          </div>
+        </form>
+      {{/hasLinkedAccountAndNoPassword}}
 
-      <div class="flex">
-        <button id="{{^isPasswordNeeded}}use-logged-in{{/isPasswordNeeded}}{{#isPasswordNeeded}}submit-btn{{/isPasswordNeeded}}" class="cta-primary cta-xl {{^isPasswordNeeded}}use-logged-in{{/isPasswordNeeded}}" type="submit">{{#t}}Sign in{{/t}}</button>
-      </div>
+      {{#hasLinkedAccountAndNoPassword}}
+        {{{ unsafeThirdPartyAuthHTML }}}
+      {{/hasLinkedAccountAndNoPassword}}
 
-        <div id="tos-pp" class="text-grey-500 my-5 text-xs">
-          {{#isPocketClient}}
-            {{#unsafeTranslate}}
-              By proceeding, you agree to:<br />
-              Pocket’s <a class="link-grey" id="pocket-tos" href="https://getpocket.com/en/tos/">Terms of Service</a> and <a id="pocket-pp" class="link-grey" href="https://getpocket.com/en/privacy/">Privacy Notice</a><br />
-              Firefox’s <a class="link-grey" id="fxa-tos" href="/legal/terms">Terms of Service</a> and <a class="link-grey" id="fxa-pp" href="/legal/privacy">Privacy Notice</a>.
-            {{/unsafeTranslate}}
-          {{/isPocketClient}}
-          {{^isPocketClient}}
-            {{#unsafeTranslate}}By proceeding, you agree to the <a class="link-grey" id="fxa-tos" href="/legal/terms">Terms of Service</a> and <a class="link-grey" id="fxa-pp" href="/legal/privacy">Privacy Notice</a>.{{/unsafeTranslate}}
-          {{/isPocketClient}}
-        </div>
 
       <div class="flex justify-between">
         <a href="/" id="use-different" class="link-blue me-2" data-flow-event="use-different-account">{{#t}}Use a different account{{/t}}</a>
-        <a href="/reset_password" class="link-blue" data-flow-event="forgot-password" id="reset-password">{{#t}}Forgot password?{{/t}}</a>
+        {{^hasLinkedAccountAndNoPassword}}
+          <a href="/reset_password" class="link-blue" data-flow-event="forgot-password" id="reset-password">{{#t}}Forgot password?{{/t}}</a>
```

Comment:
I think its safe to let the user reset their password, they would just end up setting one.